src/tokenizer/tokenize.ts

```diff
@@ -245,7 +245,26 @@ function tokenize(pattern: string, options: TokenizerOptions = {}): TokenizerRes
   };
 }
 
-function getTokenWithDetails(context: Context, pattern: string, m: string, lastIndex: number) {
+function getTokenWithDetails(context: Context, pattern: string, m: string, lastIndex: number): {
+  // Array of all of the char class's tokens
+  tokens: Array<Token>;
+  // Jump forward to the end of the char class
+  lastIndex: number; token?: undefined;
+} | {
+  token: Token;
+  tokens?: undefined;
+  // Jump forward to after the closing paren
+  lastIndex?: undefined;
+} | {
+  tokens: Array<Token>;
+  // Jump forward to after the closing paren
+  lastIndex?: undefined; token?: undefined;
+} | {
+  // Jump forward to after the closing paren
+  lastIndex: number;
+  tokens?: undefined;
+  token?: undefined;
+} {
```

Comment:
this look fun

Reply:
It's more complex than it should be; will fix in follow up